nemo/collections/nlp/data/machine_translation/one_side_dataset.py

```diff
@@ -131,3 +136,137 @@ def clean(self, ids, max_tokens=None, min_tokens=None):
+        text_tar_filepaths: str,
+        metadata_path: str,
+        tokenizer: str,
+        shuffle_n: int = 1,
+        shard_strategy: str = "scatter",
+        global_rank: int = 0,
+        world_size: int = 0,
+    ):
+        super(TarredOneSideTranslationDataset, self).__init__()
+
+        self.tokenizer = tokenizer
+        self.pad_id = tokenizer.pad_id
+
+        valid_shard_strategies = ['scatter', 'replicate']
+        if shard_strategy not in valid_shard_strategies:
+            raise ValueError(f"`shard_strategy` must be one of {valid_shard_strategies}")
+
+        with open(metadata_path, 'r') as f:
+            metadata = json.load(f)
+
+        self.metadata = metadata
+
+        if isinstance(text_tar_filepaths, str):
+            # Replace '(', '[', '<' and '_OP_' with '{'
+            brace_keys_open = ['(', '[', '<', '_OP_']
+            for bkey in brace_keys_open:
+                if bkey in text_tar_filepaths:
+                    text_tar_filepaths = text_tar_filepaths.replace(bkey, "{")
+
+            # Replace ')', ']', '>' and '_CL_' with '}'
+            brace_keys_close = [')', ']', '>', '_CL_']
+            for bkey in brace_keys_close:
+                if bkey in text_tar_filepaths:
+                    text_tar_filepaths = text_tar_filepaths.replace(bkey, "}")
+
+        if isinstance(text_tar_filepaths, str):
+            # Brace expand
+            text_tar_filepaths = list(braceexpand.braceexpand(text_tar_filepaths))
+
+        if shard_strategy == 'scatter':
+            logging.info("All tarred dataset shards will be scattered evenly across all nodes.")
+            if len(text_tar_filepaths) % world_size != 0:
+                logging.warning(
+                    f"Number of shards in tarred dataset ({len(text_tar_filepaths)}) is not divisible "
+                    f"by number of distributed workers ({world_size})."
+                )
+            begin_idx = (len(text_tar_filepaths) // world_size) * global_rank
+            end_idx = begin_idx + (len(text_tar_filepaths) // world_size)
+            logging.info('Begin Index : %d' % (begin_idx))
+            logging.info('End Index : %d' % (end_idx))
+            text_tar_filepaths = text_tar_filepaths[begin_idx:end_idx]
+            logging.info(
+                "Partitioning tarred dataset: process (%d) taking shards [%d, %d)", global_rank, begin_idx, end_idx
+            )
+
+        elif shard_strategy == 'replicate':
+            logging.info("All tarred dataset shards will be replicated across all nodes.")
+
+        else:
+            raise ValueError(f"Invalid shard strategy ! Allowed values are : {valid_shard_strategies}")
+
+        self.tarpath = text_tar_filepaths
+
+        # Put together WebDataset
+        self._dataset = (
+            wd.Dataset(text_tar_filepaths)
+            .shuffle(shuffle_n)
+            .rename(pkl='pkl', key='__key__')
+            .to_tuple('pkl', 'key')
+            .map(f=self._build_sample)
+        )
```

Comment:
I believe the syntax has changed for WebDataset with the recent version upgrade: https://github.com/NVIDIA/NeMo/pull/1760
https://github.com/NVIDIA/NeMo/pull/1760/files#diff-26041722ee3bb4aa3830321d28b90ed90eefcc10ba55292d9a7a74e0b4216281R400
Please run ./reinstall.sh or updgrade Webdataset.